web/src/echo.ts

```diff
@@ -61,28 +131,43 @@ function show_message_failed(message_id, failed_msg) {
     });
 }
 
-function show_failed_message_success(message_id) {
+function show_failed_message_success(message_id: number): void {
     // Previously failed message succeeded
     message_live_update.update_message_in_all_views(message_id, ($row) => {
         $row.find(".message_failed").toggleClass("hide", true);
     });
 }
 
-function failed_message_success(message_id) {
-    message_store.get(message_id).failed_request = false;
+function failed_message_success(message_id: number): void {
+    message_store.get(message_id)!.failed_request = false;
```

Comment:
This one could could be reworked in a follow-up to be passed the actual message object rather than looking it up, to remove this type assertion.

Reply:
Fetching `message` inside a closure(`on_success` callback) that might have an old state of message_store( the state at the time of its creation) and then passing it to `failed_message_success` might not be correct. I tried to do it and the message fetched inside the closure is `undefined`. Most likely due to the old_state that it had captured during its creation.